frappe/utils/safe_exec.py

```diff
@@ -191,6 +197,51 @@ def get_safe_globals():
 
 	return out
 
+def safe_enqueue(cmd, queue='default', job_name=None, **kwargs):
+	'''
+		Enqueue method to be executed using a background worker if not already queued
+
+		:param cmd: whitelised method or server script api method
+		:param queue: should be either long, default or short
+		:param job_name: used to identify an enqueue call, used to prevent duplicate calls
+	'''
+
+	job_name = job_name or cmd
+
+	site = frappe.local.site
+	queued_jobs = get_jobs(site=site, queue=queue, key='job_name').get(site) or []
+
+	if job_name not in queued_jobs:
+		enqueue(
+			'frappe.utils.safe_exec.execute_enqueued_cmd',
+			cmd=cmd, job_name=job_name, queue=queue, **kwargs
+		)
+
+def execute_enqueued_cmd(cmd, **kwargs):
+	'''
+		Executes in a background worker
+		Executes the method if it is a whitelisted method or server script api
+	'''
+	try:
+		# run server script if `cmd` is a server script of type API
+		is_server_script_api = run_server_script_api(cmd, kwargs)
+
+		if not is_server_script_api:
+			# update form_dict, to use as `kwargs` in frappe.call
+			frappe.form_dict.update(kwargs)
+			execute_cmd(cmd)
```

Comment:
```suggestion
			return execute_cmd(cmd)
```

Reply:
As the actual execution happens in a background worker, the return value won't be available as expected.

for eg: the return value will be `None` here ```return_value = frappe.enqueue(...)```

Reply:
Since `frappe.enqueue` returns a Job object, I think it might be required/beneficial for a lot of scenarios too. Maybe fetching info about the job through another API?

<img width="819" alt="Screenshot 2021-11-23 at 1 12 33 PM" src="https://user-images.githubusercontent.com/36654812/142986688-88055a78-8598-4eb5-9a3c-5b5af30d8614.png">

Reply:
Done!